rotkehlchen/chain/evm/decoding/uniswap/v1/decoder.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING
+
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.ethereum.modules.aave.v1.decoder import DEFAULT_DECODING_OUTPUT
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import ActionItem, DecodingOutput
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import EvmTransaction
+from rotkehlchen.utils.misc import hex_or_bytes_to_address
+
+from ..constants import CPT_UNISWAP_V1, UNISWAP_ICON, UNISWAP_LABEL
+
+if TYPE_CHECKING:
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+# https://github.com/Uniswap/v1-contracts/blob/c10c08d81d6114f694baa8bd32f555a40f6264da/contracts/uniswap_exchange.vy#L13
+TOKEN_PURCHASE = b'\xcd`\xaau\xde\xa3\x07/\xbc\x07\xaem}\x85k]\xc5\xf4\xee\xe8\x88T\xf5\xb4\xab\xf7\xb6\x80\xef\x8b\xc5\x0f'  # noqa: E501
+# https://github.com/Uniswap/v1-contracts/blob/c10c08d81d6114f694baa8bd32f555a40f6264da/contracts/uniswap_exchange.vy#L14
+ETH_PURCHASE = b'\x7f@\x91\xb4l3\xe9\x18\xa0\xf3\xaaB0vA\xd1{\xb6p)BzSi\xe5K59\x84#\x87\x05'
+
+
+class Uniswapv1CommonDecoder(DecoderInterface):
+
+    def _maybe_decode_swap(
```

Comment:
this one afaik is only ethereum. Can you check if it was deployed somewhere else?

Reply:
No, I couldn't find it anywhere else. It's only the signatures that can be matched if someone else deploys the same contract on other chains. But then, that could be some protocol other than uniswap.